lib/rdf/model/uri.rb

```diff
@@ -739,7 +739,7 @@ def ==(other)
         # If other is a Literal, reverse test to consolodate complex type checking logic
         other == self
       when String then to_s == other
-      when URI then hash == other.hash && to_s == other.to_s
+      when URI then hash == other.hash
```

Comment:
That's saying that it's possible for the hashes of two strings to be the same, which seems unlikely - that would break Hashes in Ruby pretty spectacularly, unless it's doing buckets.